tests/components/watergate/test_coordinator.py

```diff
@@ -0,0 +1,68 @@
+"""Tests for the Watergate coordinator."""
+
+from unittest.mock import AsyncMock
+
+import pytest
+from watergate_local_api import WatergateApiException
+from watergate_local_api.models import DeviceState
+from watergate_local_api.models.water_meter import WaterMeter
+
+from homeassistant.components.watergate.coordinator import (
+    WatergateAgregatedRequests,
+    WatergateDataCoordinator,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import UpdateFailed
+
+
+@pytest.fixture
+def mock_api_client():
+    """Mock WatergateLocalApiClient."""
+    client = AsyncMock()
+
+    # Simulate successful responses for each API method
+    client.async_get_device_state.return_value = DeviceState(
+        valve_state="open",
+        water_flow_indicator="on",
+        mqtt_status="True",
+        wifi_status="True",
+        power_supply="battery",
+        firmware_version="1.0.0",
+        uptime=100,
+        water_meter=WaterMeter(volume=1.2, duration=100),
+    )
+    return client
+
+
+@pytest.mark.asyncio
```

Comment:
testing internals (aka directly testing the coordinator) is something that's not liked. Rather test the effects of those things on your entities. The first test would be checked by a valve test, and for the second you can check that the valve become unavailable correctly.

Reply:
Understood. Can you, please show me how it is the best way to do it? I am struggling with a way how to check for example if valve callbacks are executed.

Reply:
which callbacks do you mean? `_handle_coordinator_update` has been correctly called if you verify that the valve has a correct state (which can for example easily be done through `snapshot` tests). The other callbacks you can verify by just calling the webhooks (for that you can check tedee `test_init` or withings again).

Reply:
Ok, sorted. Somehow.